### DISCUSSION TITLE
Very low BPM scrolls very far away when playing, skipping the next available beat line.

### DISCUSSION THREAD

**Discussion by naoei** [ppy/osu#18650] [Q&A] [answered]
A bit difficult to explain this in the title but essentially if you have a very low BPM section (e.g. 1BPM in a second), normally it scrolls fine when paused, however when the beatmap is playing, scrolling during that section will skip a large portion of the map. I expected that it would skip over until the next beat line, much like when it was paused.

An example of this can be found via video:

https://user-images.githubusercontent.com/35349837/173206174-8b3cdce4-bd07-45d9-bb81-88d2b128b7e8.mp4

**Comment by peppy** [comment, marked as answer]
The cause of this is that we intentionally increase the rate of seeking when the song is playing to be 4x as much as usual. In normal cases, this works out well (I believe stable also does this) in giving the user the seek they expect, but your edge case of 1 BPM makes it look broken.

https://github.com/ppy/osu/blob/f8830c6850128456266c82de83273204f8b74ac0/osu.Game/Screens/Edit/Editor.cs#L799-L805

I'm not sure this should be changed. Maybe it should fall back to doing a 1x seek if the BPM is below a certain number? Even at 60 BPM it still feels correct, and I wouldn't expect many use cases to drop below this.

@naoei do you actually have a mapping use case to drop the BPM that low?

**Reply by naoei** [reply]
I don't have many use cases to drop the BPM this low, but one that I can think of is "freeze a slider in place" to emphasize a freeze in the song. Though that use case is primarily used in aspire maps, I'm not entirely sure how well it would hold up in normal maps.

**Reply by peppy** [reply]
Right. That's not really valid in normal beatmaps. Slider velocity should be adjusted instead for cases like that.

**Reply by peppy** [reply]
Should be addressed by https://github.com/ppy/osu/pull/18788
